cachito/web/content_manifest.py

```diff
@@ -232,3 +238,208 @@ def generate_icm(self, image_contents=None):
         icm = deepcopy(BASE_ICM)
         icm["image_contents"] = image_contents or []
         return deep_sort_icm(icm)
+
+
+@dataclass(frozen=True)
+class Package:
+    """
+    A package within a content manifest.
+
+    It is used primarily to generate a package URL (purl).
+    """
+
+    name: str
+    type: str
+    version: str
+    dev: bool = False
+    dependencies: list = field(default_factory=list)
+    path: str = None
+
+    def __repr__(self):
+        return (
+            f"<{self.__class__.__name__} name={self.name}, type={self.type}, "
+            f"version={self.version}>"
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.type, self.version, self.dev))
+
+    def __eq__(self, other):
+        return (
+            type(other) == Package
```

Comment:
As a minor improvement, this could be `isinstance(other, Package)` or `type(other) == type(self)`. The first if the parent class should be able to compare with subclasses, the second if not. It looks like dataclasses use the second option by default.

The current code has a minor issue in that subclasses could never compare equal, even to the same subclass.
```python
class X(Package):
    pass

X("a", "b", "c") == X("a", "b", "c")  # False
```